kernel/src/main/java/org/kframework/kompile/Kompile.java

```diff
@@ -86,20 +87,32 @@
     java.util.Set<KEMException> errors;
 
     public Kompile(KompileOptions kompileOptions, FileUtil files, KExceptionManager kem, boolean cacheParses) {
-        this(kompileOptions, files, kem, new Stopwatch(kompileOptions.global), cacheParses);
+        this(kompileOptions, null, files, kem, new Stopwatch(kompileOptions.global), cacheParses);
     }
 
     public Kompile(KompileOptions kompileOptions, FileUtil files, KExceptionManager kem) {
         this(kompileOptions, files, kem, true);
     }
 
     @Inject
-    public Kompile(KompileOptions kompileOptions, FileUtil files, KExceptionManager kem, Stopwatch sw) {
-        this(kompileOptions, files, kem, sw, true);
+    public Kompile(KompileOptions kompileOptions, OuterParsingOptions outerParsing, FileUtil files, KExceptionManager kem, Stopwatch sw) {
+        this(kompileOptions, outerParsing, files, kem, sw, true);
     }
 
     public Kompile(KompileOptions kompileOptions, FileUtil files, KExceptionManager kem, Stopwatch sw, boolean cacheParses) {
+        this(kompileOptions, null, files, kem, sw, cacheParses);
+    }
+
+    public Kompile(KompileOptions kompileOptions, OuterParsingOptions outerParsing, FileUtil files, KExceptionManager kem, Stopwatch sw, boolean cacheParses) {
         this.kompileOptions = kompileOptions;
+
+        /**
+         * Invoking the kompile command will inject outerParsing into kompileOptions.outerParsing and leave the outerParsing parameter as null.
+         * Skip this assignment to prevent kompileOptions.outerParsing from being clobbered for the kompile utility.
+         */
+        if (outerParsing != null) {
+            kompileOptions.outerParsing = outerParsing;
+        }
```

Comment:
@dwightguth can you please have a look here?
On my PR you requested changes for something similar.

Reply:
I would prefer instead of mutating KompileOptions that we simply store the OuterParsingOptions in the Kompile object and reference it directly when we need to read its values. We try to avoid this type of mutability in the code base where possible.

Reply:
I think the main problem that I see is that kompileOptions.outerParsing is initialized differently depending on the command that we're running. If I can find a way to construct an instance of kompile options with the appropriate outerParsing object, we can eliminate this change along with the constructor change